app/gui/src/dashboard/components/styled/FocusArea.tsx

```diff
@@ -30,71 +29,79 @@ export interface FocusAreaProps {
   /** Should ONLY be passed in exceptional cases. */
   readonly focusDefaultClass?: string
   readonly active?: boolean
-  readonly direction: focusDirectionProvider.FocusDirection
-  readonly children: (props: FocusWithinProps) => React.JSX.Element
+  readonly direction: FocusDirection
+  readonly children: (props: FocusWithinProps) => JSX.Element
 }
 
 /** An area that can be focused within. */
 function FocusArea(props: FocusAreaProps) {
   const { active = true, direction, children } = props
   const { focusChildClass = 'focus-child', focusDefaultClass = 'focus-default' } = props
-  const { focusChildClass: outerFocusChildClass } = focusClassProvider.useFocusClasses()
-  const [areaFocus, setAreaFocus] = React.useState(false)
-  const { focusWithinProps } = aria.useFocusWithin({ onFocusWithinChange: setAreaFocus })
-  const focusManager = aria.useFocusManager()
-  const navigator2D = navigator2DProvider.useNavigator2D()
-  const rootRef = React.useRef<HTMLElement | SVGElement | null>(null)
-  const cleanupRef = React.useRef(() => {})
-  const focusChildClassRef = React.useRef(focusChildClass)
-  focusChildClassRef.current = focusChildClass
-  const focusDefaultClassRef = React.useRef(focusDefaultClass)
-  focusDefaultClassRef.current = focusDefaultClass
+  const { focusChildClass: outerFocusChildClass } = useFocusClasses()
+  const [areaFocus, setAreaFocus] = useState(false)
+  const { focusWithinProps } = useFocusWithin({ onFocusWithinChange: setAreaFocus })
+  const focusManager = useFocusManager()
+  const navigator2D = useNavigator2D()
+  const rootRef = useRef<HTMLElement | SVGElement | null>(null)
+  const cleanupRef = useRef(() => {})
+  const focusChildClassRef = useSyncRef(focusChildClass)
+  const focusDefaultClassRef = useSyncRef(focusDefaultClass)
+
+  useSingleUnmount(() => {
```

Comment:
Well, I don't think suppressing StrictMode is good, because usually when we need to do that, it indicates buggy code.

Reply:
React 19 fires ref callbacks twice in strict mode as well :)

Reply:
@MrFlashAccount yeah, but that's fine - because then the entire body fires twice, right? our issue is that ref callbacks happen *once* but cleanup happens *twice* i think. so it gets cleaned up immediately before anything happens which is not what we want

(otherwise i don't think i would have needed this in the first place?)

although it *is* called with `null` on unmount so that should be reliable enough i guess.

Reply:
It's unusual, Cleanup in StrictMode happens once. So the flow is body -> cleanup -> body

Maybe we have an unmountEffect somewhere?